doc/source/flexibility/binlogging_replication_improvements.rst

```diff
@@ -0,0 +1,175 @@
+excessive precaution: once some statement with temporary table had appeared and
+the row-based logging was used, |MySQL| logged unconditionally all
+subsequent statements in row format.
+
+Percona Server have implemented more accurate behavior: instead of switching to
+row-based logging until the last temporary table is closed, the usual rules of
+row vs statement format apply, and presence of currently opened temporary
+tables is no longer considered. This change was introduced with the fix of a
+bug :psbug:`151` (upstream :mysqlbug:`72475`).
+
+Temporary table drops and binloging on GTID-enabled server
+==========================================================
+
+Summary of the fix:
+*******************
+
+MySQL logs DROP statements for all temporary tables irrelative of the logging
+mode under which these tables were created. This produces binlog writes and
+errand GTIDs on slaves with row and mixed logging. |Percona Server| fixes this
+by tracking the binlog format at temporary table create time and using it to
+decide whether a DROP should be logged or not.
+
+Version Specific Information
+****************************
+
+  * :rn:`5.7.17-11`
+    Fix ported from |Percona Server| 5.6
+
+Details:
+********
+
+Even with read_only mode enabled, the server permits some operations, including
+ones with temporary tables. With the previous fix, temporary table operations
+are not binlogged in row or mixed mode. But |MySQL| doesn’t track what was
+the logging mode when temporary table was created, and therefore
+unconditionally logs ``DROP`` statements for all temporary tables. These
+``DROP`` statements receive ``IF EXISTS`` addition, which is intended to make
+them harmless.
+
+|Percona Server| have fixed this with the bug fixes :psbug:`964`, upstream
+:mysqlbug:`83003`, and upstream :mysqlbug:`85258`. Moreover, after all the
+binlogging fixes discussed so far nothing involving temporary tables is logged
+to binary log in row or mixed format, and so there is no need to consider
+``CREATE/DROP TEMPORARY TABLE`` unsafe for use in stored functions, triggers,
+and multi-statement transactions in row/mixed format. Therefore an additional
+fix was introduced to mark creation and drop of temporary tables as unsafe
+inside transactions in statement-based replication only (bug fixed
+:psbug:`1816`, upstream :mysqlbug:`89467`)).
+
+Safety of statements with a ``LIMIT`` clause
+============================================
+
+Summary of the fix:
+*******************
+
+|MySQL| considers all ``UPDATE/DELETE/INSERT ... SELECT`` statements with
+``LIMIT`` clause to be unsafe, no matter wether they are really producing
+non-deterministic result or not, and switches from statement-based logging
+to row-based one. |Percona Server| is more accurate, it acknowledges such
+instructions as safe when they include ``ORDER BY PK`` or ``WHERE``
+condition. This fix has been ported from the upstream bug report
+:mysqlbug:`42415`.
```

Comment:
Removed mention of PS-44 issue, because it is internal.

Reply:
It was not correct for PS-44 to be internal - I have sanitized it and made public - please restore.